CMakeLists.txt

```diff
@@ -290,6 +290,15 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
   target_compile_options(ds2 PRIVATE /MP)
 endif ()
 
+if (COVERAGE AND (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"))
+  target_compile_options(ds2 PRIVATE -fprofile-arcs -ftest-coverage)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
```

Comment:
The way you set these flags won't work for some stuff in JSObjects. Let's talk about it before checking it in.